src/nodal_properties.cc

```diff
@@ -0,0 +1,30 @@
+#include "nodal_properties.h"
+
+// Function to create new property with given name and size (rows x cols)
+void mpm::NodalProperties::create_property(const std::string& property,
+                                           unsigned rows, unsigned columns) {
+  // Initialize a matrix with size of rows times columns and insert it to the
+  // property database map
+  Eigen::MatrixXd property_data = Eigen::MatrixXd::Zero(rows, columns);
+  properties_.insert(
+      std::pair<std::string, Eigen::MatrixXd>(property, property_data));
+}
+
+// Return data in the nodal properties map at a specific index
+Eigen::MatrixXd mpm::NodalProperties::property(const std::string& property,
+                                               unsigned node_id,
+                                               unsigned mat_id,
+                                               unsigned nprops) const {
+  double* position = &properties_.at(property)(node_id * nprops, mat_id);
+  mpm::MapProperty property_map(position, nprops);
+  return property_map;
+}
+
+// Assign property value to a pair of node and material
+void mpm::NodalProperties::assign_property(const std::string& property,
+                                           unsigned node_id, unsigned mat_id,
+                                           Eigen::MatrixXd property_value,
+                                           unsigned nprops) {
+  properties_.at(property).block(node_id * nprops, mat_id, nprops, 1) =
```

Comment:
What does the `1` at the end denote?

Reply:
The two last arguments are an indication of the size of the block. nprops is the number of rows and 1 is the number of columns. Since only one material is returned, 1 is always the argument passed in this case.

Reply:
Please add that in the comment